corehq/apps/motech/openmrs/concepts/sync.py

```diff
@@ -0,0 +1,112 @@
+import json
+import jsonobject
+from corehq.apps.motech.openmrs.concepts.models import OpenmrsConcept
+from corehq.apps.motech.openmrs.restclient.listapi import OpenmrsListApi
+
+
+def sync_concepts_from_openmrs(account):
+    requests = account.get_requests_object()
+    OpenmrsConcept.objects.filter(account=account).delete()
+    api = OpenmrsListApi(requests, 'concept')
+    answers_relationships = []
+    for concept in api.get_all():
+        concept = openmrs_concept_json_from_api_json(concept)
+        concept, answers = openmrs_concept_from_concept_json(account, concept)
+        if answers:
+            answers_relationships.append((concept, answers))
+        concept.save()
+
+    while answers_relationships:
+        delayed = []
+        for concept, answers in answers_relationships:
+            answer_concepts = OpenmrsConcept.objects.filter(account=account, uuid__in=answers).all()
+            if set(answer_concept.uuid for answer_concept in answer_concepts) == set(answers):
+                concept.answers = answer_concepts
+                concept.save()
+            else:
+                delayed.append((concept, answers))
+
+        if len(answers_relationships) == len(delayed):
+            # this is going to be an infinite loop
+            raise Exception(delayed)
```

Comment:
Actually I think this approach isn't necessary at all—at one point I think I was trying to get each concept to save only once, so I basically had to feel out the dependency graph using that method. Now that they're all saved up front and then they're re-saved after adding answers (which may only actually update the concept-to-answer table) I think it's no longer necessary.